cmake/WarningsErrorsPolicy.cmake

```diff
@@ -0,0 +1,24 @@
+set(warnings shadow parentheses sign-compare unused-variable)
```

Comment:
Might be worth making this configurable for when we want to enable additional warnings?

Reply:
I think adding is fine.